src/crypto/CHIPCryptoPAL.h

```diff
@@ -1492,6 +1492,37 @@ CHIP_ERROR ExtractSKIDFromX509Cert(const ByteSpan & certificate, MutableByteSpan
  **/
 CHIP_ERROR ExtractAKIDFromX509Cert(const ByteSpan & certificate, MutableByteSpan & akid);
 
+/**
+ * @brief Extracts the Authority Key Identifier from an X509 Certificate.
+ **/
+CHIP_ERROR ExtractAKIDFromX509Cert(const ByteSpan & certificate, MutableByteSpan & akid);
+
+/**
+ * @brief Checks for resigned version of the certificate in the list and returns it.
+ *
+ * The following three conditions SHOULD be satisfied for the certificate to qualify as
+ * a resigned version of a reference certificate:
+ *   - SKID of the candidate and the reference certificate should match.
+ *   - SubjectDN of the candidate and the reference certificate should match.
+ *   - NotBefore of the candidate certificate should be greater or equal than the
```

Comment:
Not sure we should validate this last condition. The certs in the replacement set can just be made to have valid dates. In both the case where the notbefore is checked and not checked, providing erroneous certs would make the substitution fail. Not accounting for trying to validate the notbefore date would simplify the code.

Reply:
I agree, for simplicity I removed the time validity checks and the comparison will be based only on SKID and Subject.